pymc_experimental/model/transforms/autoreparam.py

```diff
@@ -0,0 +1,375 @@
+            \end{cases}
+
+
+
+        Parameters
+        ----------
+        value : float
+            :math:`\varepsilon`
+        """
+        truncate = dict.fromkeys(
+            self._logit_lambda.keys(),
+            value,
+        )
+        self.truncate_lambda(**truncate)
+
+    def get_lambda(self) -> Dict[str, np.ndarray]:
+        r"""Get :math:`\lambda_k` that are currently used by the model.
+
+        Returns
+        -------
+        Dict[str, np.ndarray]
+            Mapping from variable name to :math:`\lambda_k`.
+        """
+        return {
+            name: scipy.special.expit(shared.get_value())
+            for name, shared in self._logit_lambda.items()
+        }
+
+    def set_lambda(self, **kwargs: Dict[str, Union[np.ndarray, float]]):
+        r"""Set :math:`\lambda_k` per variable."""
+        for key, value in kwargs.items():
+            logit_lam = scipy.special.logit(value)
+            shared = self._logit_lambda[key]
+            fill = np.broadcast_to(
+                logit_lam,
+                shared.type.shape,
+            )
+            shared.set_value(fill)
+
+    def set_all_lambda(self, value: Union[np.ndarray, float]):
+        r"""Set :math:`\lambda_k` globally."""
+        config = dict.fromkeys(
+            self._logit_lambda.keys(),
+            value,
+        )
+        self.set_lambda(**config)
+
+    def fit(self, *args, **kwargs) -> pm.Approximation:
+        r"""Set :math:`\lambda_k` using Variational Inference.
+
+        Examples
+        --------
+
+        .. code-block:: python
+
+            with model:
+                # set all parameterizations to mix of centered and non-centered
+                vip.set_all_lambda(0.5)
+
+                # fit using ADVI
+                mf = vip.fit(random_seed=42)
+        """
+        kwargs.setdefault("obj_optimizer", pm.adagrad_window(learning_rate=0.1))
+        kwargs.setdefault("method", "advi")
+        return pm.fit(
+            *args,
+            more_obj_params=self.variational_parameters,
+            **kwargs,
+        )
+
+
+def vip_reparam_node(
+    op: RandomVariable,
+    node: Apply,
+    name: str,
+    dims: List[Variable],
+    transform: Transform,
+) -> Tuple[ModelDeterministic, ModelNamed]:
+    if not isinstance(node.op, RandomVariable):
+        raise TypeError("Op should be RandomVariable type")
+    size = node.inputs[1]
+    if not isinstance(size, pt.TensorConstant):
+        raise ValueError("Size should be static for autoreparametrization.")
+    logit_lam_ = pytensor.shared(
+        np.zeros(size.data),
+        shape=size.data,
+        name=f"{name}::lam_logit__",
+    )
+    logit_lam = model_named(logit_lam_, *dims)
+    lam = pt.sigmoid(logit_lam)
+    return (
+        _vip_reparam_node(
+            op,
+            node=node,
+            name=name,
+            dims=dims,
+            transform=transform,
+            lam=lam,
+        ),
+        logit_lam,
+    )
+
+
+@singledispatch
```

Comment:
Seems a bit overkill at this point?

Reply:
I'd extend it to student T, Gamma, Logit Normal, etc